File: app/src/main/java/com/xianzhitech/ptt/AppComponent.java
package com.xianzhitech.ptt;

import com.squareup.okhttp.OkHttpClient;
import com.xianzhitech.ptt.engine.TalkEngineProvider;
import com.xianzhitech.ptt.service.provider.AuthProvider;
import com.xianzhitech.ptt.service.provider.SignalProvider;

/**
 *
 * 应用程序组件. 提供各种库依赖
 *
 * Created by fanchao on 13/12/15.
 */
public interface AppComponent {
    OkHttpClient providesHttpClient();

    SignalProvider providesSignal();

    TalkEngineProvider providesTalkEngine();
    Database providesDatabase();
    Broker providesBroker();
    AuthProvider providesAuth();
}

File: app/src/main/kotlin/com/xianzhitech/ptt/AppComponent.kt
package com.xianzhitech.ptt

import com.fasterxml.jackson.databind.ObjectMapper
import com.google.gson.Gson
import com.xianzhitech.ptt.api.AppApi
import com.xianzhitech.ptt.api.SignalApi
import com.xianzhitech.ptt.broker.SignalBroker
import com.xianzhitech.ptt.data.Storage
import com.xianzhitech.ptt.media.MediaButtonHandler
import com.xianzhitech.ptt.repo.*
import com.xianzhitech.ptt.service.AppService
import com.xianzhitech.ptt.service.handler.SignalServiceHandler
import com.xianzhitech.ptt.service.handler.StatisticCollector
import com.xianzhitech.ptt.ui.ActivityProvider
import okhttp3.OkHttpClient

/**

 * 应用程序组件. 提供各种库依赖

 * Created by fanchao on 13/12/15.
 */
interface AppComponent {
    val httpClient: OkHttpClient
    val preference: Preference
    val gson: Gson

    val storage : Storage
    val objectMapper : ObjectMapper

    val appApi : AppApi
    val signalBroker : SignalBroker

    val appServerEndpoint : String
    val appService: AppService
    val signalHandler: SignalServiceHandler

    val mediaButtonHandler: MediaButtonHandler

    // Repositories
    val userRepository: UserRepository
    val groupRepository: GroupRepository
    val roomRepository: RoomRepository
    val contactRepository: ContactRepository
    val messageRepository : MessageRepository

    val statisticCollector: StatisticCollector
    val activityProvider: ActivityProvider
}
